tests/_test_msui/test_multiple_flightpath_dockwidget.py

```diff
@@ -25,33 +25,230 @@
     limitations under the License.
 """
 import pytest
-from PyQt5 import QtTest
+from PyQt5 import QtWidgets, QtCore, QtGui, QtTest
+from unittest import mock
+
 from mslib.msui import msui
-from mslib.msui.multiple_flightpath_dockwidget import MultipleFlightpathControlWidget
-from mslib.msui import flighttrack as ft
 import mslib.msui.topview as tv
 
 
-class Test_MultipleFlightpathControlWidget:
-    @pytest.fixture(autouse=True)
-    def setup(self, qtbot):
-        self.window = msui.MSUIMainWindow()
-        self.window.create_new_flight_track()
-
-        self.window.actionNewFlightTrack.trigger()
-        self.window.actionTopView.trigger()
-        initial_waypoints = [ft.Waypoint(40., 25., 0), ft.Waypoint(60., -10., 0), ft.Waypoint(40., 10, 0)]
-        self.waypoints_model = ft.WaypointsTableModel("myname")
-        self.waypoints_model.insertRows(
-            0, rows=len(initial_waypoints), waypoints=initial_waypoints)
-
-        self.widget = tv.MSUITopViewWindow(model=self.waypoints_model, mainwindow=self.window)
-        self.window.show()
-        QtTest.QTest.qWaitForWindowExposed(self.window)
-        yield
-        self.window.hide()
-
-    def test_initialization(self):
-        widget = MultipleFlightpathControlWidget(parent=self.widget,
-                                                 listFlightTracks=self.window.listFlightTracks)
-        assert widget.color == (0, 0, 1, 1)
+@pytest.fixture
+def main_window(qtbot):
+    """
+    Set-up for the docking widget
+    """
+    # Start a MSUI window
+    window = msui.MSUIMainWindow()
+    window.show()
+    qtbot.wait_exposed(window)
+
+    # Create two flight tracks
+    window.actionNewFlightTrack.trigger()
+    window.actionNewFlightTrack.trigger()
+
+    # Open a Top View window
+    window.actionTopView.trigger()
+    topview_window = window.listViews.currentItem().window
+
+    # Switch to the Multiple Flightpath Widget
+    topview_window.cbTools.setCurrentIndex(6)
+
+    # Get a reference to the created MultipleFlightpathControlWidget
+    multiple_flightpath_widget = topview_window.docks[tv.MULTIPLEFLIGHTPATH].widget()
+
+    yield window, multiple_flightpath_widget
+
+    window.hide()
+
+
+def test_initialization(main_window):
+    """
+    test for conforming docking widget has initialized
+    """
+    _, multiple_flightpath_widget = main_window
+
+    # Ensure the MultipleFlightpathControlWidget is correctly initialized
+    assert multiple_flightpath_widget is not None
+    assert multiple_flightpath_widget.color == (0, 0, 1, 1)
+
+
+def test_setColour(main_window):
+    """
+    test for the filghttrack colour
+    """
+    main_window, multiple_flightpath_widget = main_window
+    color_button = multiple_flightpath_widget.pushButton_color
+
+    # Mock the exec_ method to accept the dialog otherwise test get stuck
+    with mock.patch("mslib.utils.colordialog.CustomColorDialog.exec_",
```

Comment:
Even this mock shouldn't be necessary. I think it should be possible to programmatically follow the entire flow of selecting a color doing this:
```python
# Activate the first flight track
activate_flight_track_at_index(main_window, 0)

# Click on the second flight track in the docking widget
click_on_flight_track_in_docking_widget_at_index(multiple_flightpath_widget, 1)

# Simulate clicking the button to open the color dialog
color_button.click()

# Get a reference to the custom color dialog
color_dialog = multiple_flightpath_widget.findChild(QtWidgets.QDialog)
assert color_dialog is not None

# Select the first color
color = QtGui.QColor(color_dialog.colors[0])
color_dialog.color_buttons[0].click()

# Verify that the flight track data structure was updated with the new color
wp_model = multiple_flightpath_widget.list_flighttrack.currentItem().flighttrack_model
applied_color_rgba = multiple_flightpath_widget.get_color(wp_model)

# Convert the applied_color_rgba to a QColor object
applied_color = QtGui.QColor.fromRgbF(*applied_color_rgba)

assert applied_color == color
```

This just requires some minor adjustments in other parts:
```patch
diff --git a/mslib/msui/multiple_flightpath_dockwidget.py b/mslib/msui/multiple_flightpath_dockwidget.py
index 380fb6bc..bdaf9cfa 100644
--- a/mslib/msui/multiple_flightpath_dockwidget.py
+++ b/mslib/msui/multiple_flightpath_dockwidget.py
@@ -422,7 +422,7 @@ class MultipleFlightpathControlWidget(QtWidgets.QWidget, ui.Ui_MultipleViewWidge
                 else:
                     color_dialog = CustomColorDialog(self)
                     color_dialog.color_selected.connect(lambda color: self.apply_color(wp_model, color))
-                    color_dialog.exec_()
+                    color_dialog.show()
             else:
                 self.labelStatus.setText("Status: Check Mark the flighttrack to change its color.")
         elif self.list_operation_track.currentItem() is not None:
@@ -983,7 +983,7 @@ class MultipleFlightpathOperations:
                 else:
                     color_dialog = CustomColorDialog(self.parent)
                     color_dialog.color_selected.connect(lambda color: self.apply_color(op_id, color))
-                    color_dialog.exec_()
+                    color_dialog.show()
             else:
                 self.parent.labelStatus.setText("Status: Check Mark the flighttrack to change its color.")
 
diff --git a/mslib/utils/colordialog.py b/mslib/utils/colordialog.py
index 53ac19f2..6a2a0f0f 100644
--- a/mslib/utils/colordialog.py
+++ b/mslib/utils/colordialog.py
@@ -60,6 +60,7 @@ class CustomColorDialog(QtWidgets.QDialog):
             row = i // 5
             col = i % 5
             swatch_layout.addWidget(button, row, col)
+            self.color_buttons.append(button)
 
         # Add "Pick Custom Color" button
         self.custom_color_button = QtWidgets.QPushButton("Pick Custom Color")
```

Adding the buttons to self.color_buttons is something that I presume was just overlooked anyway, since the list exists but is unused so far. The change from exec_ to show is a more appropriate use of the widget IMO, since exec_ blocks further execution and show doesn't. The former is only really the right choice when the return value of exec_ is then used, but in this case the dialog communicates via signals, which is an asynchronous style compatible with show.